component/authenticator/src/main/java/org/wso2/carbon/identity/application/authenticator/totp/TOTPAuthenticator.java

```diff
@@ -201,6 +206,33 @@ protected void initiateAuthenticationRequest(HttpServletRequest request,
             if (context.isRetrying()) {
                 retryParam = "&authFailure=true&authFailureMsg=login.fail.message";
             }
+            IdentityErrorMsgContext errorContext = IdentityUtil.getIdentityErrorMsg();
+            IdentityUtil.clearIdentityErrorMsg();
+
+            String errorParam = StringUtils.EMPTY;
+            if (Boolean.parseBoolean(showAuthFailureReasonOnLoginPage)) {
+                if (errorContext != null && errorContext.getErrorCode() != null) {
+                    log.debug("Identity error message context is not null");
+                    String errorCode = errorContext.getErrorCode();
```

Comment:
can `errorCode` be null? if so better to do a null check

Reply:
Addressed in 917a93c15aa7fb493b77d849f62205caa2623e89